### PR TITLE
feat: Add image deprecation_time

### PR DESCRIPTION
**Pull request by nndegz**
#### Summary

---

Use the following steps to ensure your PR is ready to be reviewed

- [x] Read the [contribution guidelines](../blob/main/CONTRIBUTING.md) 🧑‍🎓
- [x] Run `go fmt` to format your code 🖊
- [x] Lint your changes via `golangci-lint run` 🚨 (install golangci-lint [here](https://golangci-lint.run/usage/install/#local-installation))
- [x] Update or add tests. Learn more about testing [here](https://docs.cloudquery.io/docs/developers/sdk/testing) 🧪
- [x] Update the docs by running `go run ./docs/docs.go` and committing the changes 📃
- [x] If adding a new resource, add [relevant Terraform files](../terraform) in a separate PR 📂
- [x] Ensure the status checks below are successful ✅

### CODE REVIEW

File: resources/services/ec2/images_mock_test.go [hunk 1]
```diff
@@ -26,6 +26,8 @@ func buildEc2ImagesMock(t *testing.T, ctrl *gomock.Controller) client.Services {
 	creationDate := "1994-11-05T08:15:30-05:00"
 	g.OwnerId = aws.String("testAccount")
 	g.CreationDate = &creationDate
+	deprecationTime := "2050-11-05T08:15:30-05:00"
```
Comment (erezrokah) [inline resources/services/ec2/images_mock_test.go:29]: @nndegz can you confirm this is the expect format of `DeprecationTime`?

**Review by erezrokah — DISMISSED** [review]
Thanks for the PR @nndegz 🚀  I added another commit to address the test failure, but will need your help to verify (see comment)

**Review by bbernays — APPROVED** [review]

**Pull request merged**